src/components/AuthFooter/AuthFooter.tsx

```diff
@@ -54,8 +56,36 @@ function getOAuthLink(
       : ""
   }`;
 
+  const oAuthClicked = useRef(false);
+  const category = isOauthSignup ? CATEGORY.SIGNUP : CATEGORY.SIGNIN;
+
+  useEffect(() => {
+    const listener = () =>
+      window.addEventListener("beforeunload", () => {
+        if (oAuthClicked.current) {
+          telemetry.track({
+            event: EVENT.OAUTH_CLICKED,
+            properties: {
+              category,
+              oauthProvider: oAuthProvider.name
+            }
+          });
+        }
+      });
+    listener();
+
+    return () => window.removeEventListener("beforeunload", listener);
```

Comment:
SO... `listener` is a function that then adds the event listener, not the listener itself. As such, I don't think it'll actually work with the `removeEventListener` function. I don't think you need the function at all.
```suggestion
    const listener = window.addEventListener("beforeunload", () => {
      if (oAuthClicked.current) {
        telemetry.track({
          event: EVENT.OAUTH_CLICKED,
          properties: {
            category,
            oauthProvider: oAuthProvider.name
          }
        })
      }
    })

    return () => window.removeEventListener("beforeunload", listener);
```

Reply:
![Screen Shot 2022-05-03 at 10 28 26 AM](https://user-images.githubusercontent.com/80717697/166484743-f5b79333-7d7f-44ab-ae8b-c67678f19597.png)
looks like listener has to be a function? @knagurski

Reply:
Sorry, I'm a complete numpty (technical term). Try this...
```suggestion
    const listener = () => {
      if (oAuthClicked.current) {
        telemetry.track({
          event: EVENT.OAUTH_CLICKED,
          properties: {
            category,
            oauthProvider: oAuthProvider.name
          }
        })
      }
    }
    
    window.addEventListener('beforeunload', listener)

    return () => window.removeEventListener('beforeunload', listener)
```

So hard to code in a textbox!

Reply:
updated